src/ast/ddl.rs

```diff
@@ -256,11 +263,38 @@ impl fmt::Display for TableConstraint {
                 name,
                 columns,
                 is_primary,
+                is_mysql_unique_key,
```

Comment:
This is a bad design that can turn this code huge when considering other dialects.

If you `really` need the dialect info here, maybe you could have the `Dialect` itself, not that level of specificity here.

Since there are 3 possible options, couldn't we use an enum variant? Like:

```rust
pub enum UniqueType {
    PrimaryKey,
    Key,
    Simple
} 
```

(ignore the horrible naming, but you got the idea hehe)

Reply:
Use enum variant would not work, i think using specific dialect is a good solution, but when I add dialect trait in TableConstrait struct, the compiler says: the trait bound `dyn dialect::Dialect: Clone` is not satisfied。
There may not have explict `constrait` key word in mysql create statement, `constraint` keyword is optional, like:
```
CREATE TABLE table_name(
   column_name1 column_definition,
   column_name2 column_definition,
   ...,
   [CONSTRAINT constraint_name]
      UNIQUE(column_name1,column_name2)
);
```
 so use the origin `display_constraint_name` method not work，This is why I use `is_mysql_unique_key` to distinguish mysql dialect specifically.

I haven't been able to find an elegant way to solve it yet, any good idea?

Reply:
> Use enum variant would not work

Why would this not work?

Reply:
If I remove `is_mysql_unique_key` and use the original display method, then run the parse_table_colum_option_key test, it would panic, the painc message show the difference:

thread parse_table_colum_option_key panicked at assertion failed: (left == right)
  left: "CREATE TABLE foo (modification_time DATETIME ON UPDATE, **UNIQUE KEY u_m_t (modification_time))**",
 right: "CREATE TABLE foo (modification_time DATETIME ON UPDATE, **CONSTRAINT u_m_t UNIQUE (modification_time))**", src/test_utils.rs:95:13

Reply:
@RainJoe but you can still use an enum or something like that. Using a single boolean makes the expansion difficult, and doesn't make that much sense as it is really specific information added to the general AST. Using an enum there allows the upper levels to match and handle as desired when, for example, they're using the PostgreSQL dialect.

Reply:
@AugustoFKL Thanks for your suggestion，I will make some modifications and create a now PR later.